src/litdata/streaming/dataset.py

```diff
@@ -542,6 +547,8 @@ def _replay_chunks_sampling(
             if indexes[worker_idx] >= size:
                 indexes[worker_idx] -= size
                 chunks_index[worker_idx] += 1
+                # TODO: find robust soln, as it only seems to work for 1 worker
+                # chunks_index[worker_idx] = (chunks_index[worker_idx] + 1) % len(intervals)
```

Comment:
Currenlty, this only seems to work properly incase of 1 worker

Reply:
Incase of num_workers>1, the dataloader only outputs certain batches

Reply:
After further going through the steps, it seems the index error was caused by different items, but it originated because, in the fresh epoch, it is receiving num_samples_yielded as the full dataset length instead of 0. To fix this, the dataset reset state is required.